src/components/ItemList/Item.jsx

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+
+function Item({ item, onChange, onDelete }) {
+  const [isEditing, setIsEditing] = useState(false);
+  let itemContent;
+  if (isEditing) {
+    itemContent = (
+      <>
+        <input
+          value={item.text}
+          onChange={(e) => {
```

Comment:
it would be better to have a text piece of state that the onChange updates instead of updating state with every letter the user types.